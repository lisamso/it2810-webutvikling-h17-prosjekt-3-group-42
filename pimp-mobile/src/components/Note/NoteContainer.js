import React, {Component} from 'react'
import { Text, TextInput, TouchableNativeFeedback, View} from "react-native";
import Modal from 'react-native-modal';
import Note from "./Note";
import { styles } from '../../styles.js';

class NoteContainer extends Component {

  constructor(props) {
    super(props);
    this.state = {
      notes: [],
      modalVisible: false,
      title: '',
      text: '',
      count: 0
    };
  }

  componentDidMount() {
    this.newNote('Example', 'Press new note to make a new note');
  }

  toggleModal() {
    this.setState((prevState) => {
      return {modalVisible: !prevState.modalVisible};
    });
  }

  submitNote() {
    if (this.state.title !== '') {
      this.newNote(this.state.title, this.state.text);
    }
    this.setState({title: '', text: ''});
    this.toggleModal()
  }

  removeNote(id) {
    let arr = this.state.notes;
    let index = -1;
    for (let i = 0; i < arr.length; i++) {
      if (arr[i].props.id === id) {
        index = i;
      }
    }
    if (index !== -1) {
      arr.splice(index, 1);
    }
    this.setState({notes: arr});
  }

  newNote(title='',text=''){
    let note =
      <Note
        id={this.state.count}
        title={title}
        text={text}
      />;
    this.setState((prevState) => {
      return {count: prevState.count + 1}
    });
    let arr = this.state.notes;
    arr.push(note);
    this.setState({ notes: arr })
  }

  allNotes() {
    return(
      this.state.notes.map((n,idx) =>
        <View style={styles.row_box} key={idx}>
          <View style={[styles.container, styles.carrot_border]}>
            {n}
          </View>
          <TouchableNativeFeedback
            onPress={() => { this.removeNote(n.props.id)}}
          >
            <View style={styles.delete_container}>
              <View style={[styles.delete_border, styles.carrot_border]}>
                <Text style={[styles.delete_text, styles.carrot]}>x</Text>
              </View>
            </View>
          </TouchableNativeFeedback>
        </View>
      ));
  }

  render(){
    return(
      <View style={styles.box}>
        <View style={[styles.box_header, styles.carrot_border]}>
          <Text style={[styles.title, styles.carrot]}>Your Notes</Text>
        </View>
        <View style={[styles.box_body, styles.carrot_border]}>
          {this.allNotes()}
        </View>
        <TouchableNativeFeedback
          onPress={() => { this.toggleModal()}}>
          <View style={[styles.box_button, styles.carrot_background]}>
            <Text style={styles.wet_asphalt}>
              New note
            </Text>
          </View>
        </TouchableNativeFeedback>

        <Modal
          animationIn="slideInLeft"
          animationOut="slideOutRight"
          transparent={true}
          isVisible={this.state.modalVisible}
          backDropColor="white"
          onBackdropPress={() => {this.toggleModal()}}
          onBackButtonPress={() => {this.toggleModal()}}
        >
          <View style={styles.box}>
            <View style={[styles.box_header, styles.carrot_border]}>
              <Text style={[styles.title, styles.carrot]}>New note</Text>
            </View>
            <View style={[styles.box_body, styles.carrot_border]}>
              <TextInput
                style={{height: 60, color: '#e67e22'}}
                placeholder="Title"
                onChangeText={(text) => this.setState({title: text})}
              />
              <TextInput
                style={{height: 60, color: '#e67e22'}}
                placeholder="Text"
                onChangeText={(text) => this.setState({text: text})}
              />
            </View>
            <TouchableNativeFeedback
              onPress={() => {this.submitNote()}}>
              <View style={[styles.box_button, styles.carrot_background]}>
                <Text style={styles.wet_asphalt}>
                    Confirm
                </Text>
              </View>
            </TouchableNativeFeedback>
          </View>
        </Modal>
      </View>

    )
  }
}

NoteContainer.propTypes = {

}


export default NoteContainer

