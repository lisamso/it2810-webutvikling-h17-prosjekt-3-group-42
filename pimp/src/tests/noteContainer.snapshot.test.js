import NoteContainer from '../components/Note/NoteContainer';
import React from 'react';
import renderer from 'react-test-renderer';

test('NoteContainer component renders the note correctly', () => {
    //const note = {title: 'Alpacas', text: 'are beautiful', lastChanged: '12.10.2017'}
    const rendered = renderer.create(
        <NoteContainer/>
    );
    let tree = rendered.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger btnClick
    rendered.getInstance().btnClick();
    tree = rendered.toJSON();
    expect(tree).toMatchSnapshot();
});