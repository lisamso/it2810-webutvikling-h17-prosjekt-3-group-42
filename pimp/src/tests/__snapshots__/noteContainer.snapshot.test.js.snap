// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteContainer component renders the note correctly 1`] = `
<div
  className="container-item"
  id="Notes"
>
  <div
    className="box undefined"
  >
    <h2
      id="box-title"
    >
      Your notes
    </h2>
    <hr />
    <div
      className="box-container"
    />
    <div
      id="btn"
    >
      <button
        onClick={[Function]}
      >
        New note
      </button>
    </div>
  </div>
</div>
`;

exports[`NoteContainer component renders the note correctly 2`] = `
<div
  className="container-item"
  id="Notes"
>
  <div
    className="box undefined"
  >
    <h2
      id="box-title"
    >
      Your notes
    </h2>
    <hr />
    <div
      className="box-container"
    >
      <div
        className="box-item"
      >
        <div
          className="note"
        >
          <form
            className="note-form"
            onSubmit={[Function]}
          >
            <div
              className="group"
            >
              <input
                name="title"
                onChange={[Function]}
                placeholder="Title"
                required={true}
                type="text"
                value=""
              />
              <input
                type="submit"
                value="submit"
              />
              <span
                className="bar"
              />
            </div>
            <div
              className="group"
            >
              <input
                name="content"
                onChange={[Function]}
                placeholder="Content"
                required={true}
                type="text"
                value=""
              />
              <span
                className="bar"
              />
            </div>
          </form>
          <div
            id="created-at"
          >
            Press enter to save
          </div>
        </div>
      </div>
    </div>
    <div
      id="btn"
    >
      <button
        onClick={[Function]}
      >
        New note
      </button>
    </div>
  </div>
</div>
`;
